lib/api.ts

```diff
@@ -139,18 +158,31 @@ module.exports = function (PersistObjectTemplate, baseClassForPersist) {
          * @param {object} logger objecttemplate logger
          * @returns {object}
          */
-        template.getFromPersistWithId = async function(id, cascade, isTransient, idMap, isRefresh, logger) {
-            (logger || PersistObjectTemplate.logger).debug({component: 'persistor', module: 'api', activity: 'getFromPersistWithId',
-                data: {template: template.__name__, id: id}});
+        template.getFromPersistWithId = async function (id, cascade, isTransient, idMap, isRefresh, logger) {
+            (logger || PersistObjectTemplate.logger).debug({
+                component: 'persistor', module: 'api', activity: 'getFromPersistWithId',
+                data: { template: template.__name__, id: id }
+            });
             var dbType = PersistObjectTemplate.getDB(PersistObjectTemplate.getDBAlias(template.__collection__)).type;
+            const time = getTime();
             let getQuery = (dbType == PersistObjectTemplate.DB_Mongo ?
                 PersistObjectTemplate.getFromPersistWithMongoId(template, id, cascade, isTransient, idMap, logger) :
                 PersistObjectTemplate.getFromPersistWithKnexId(template, id, cascade, isTransient, idMap, isRefresh, logger))
-                .then(function(res) {
+                .then(function (res) {
                     return res;
                 }.bind(this));
 
-            return getQuery.catch(e => logExceptionAndRethrow(e, logger || PersistObjectTemplate.logger, template.__name__, id, 'getFromPersistWithId'));
+            const name = 'getFromPersistWithId';
+            return getQuery
```

Comment:
if these are all async functions, should we use async / await pattern since this is mostly new code being added? question applies for all instances

Reply:
Yeah I thought of that, but that would be a much larger feature set and change I think, because we'd have to de-promise the rest of the code (and convert to async / await). Otherwise, we'd have two paradigms in the same function (1 that uses await for the new code, and 1 that uses the old way for the older persistor functionality). It seems best to keep it uniform so we can easily convert in the future. I can create an issue here to handle conversion in the future